mysql/infoschema_test.go

```diff
@@ -0,0 +1,306 @@
+			rows:  [][]driver.Value{{"id", "PRIMARY KEY"}},
+		},
+	}
+	db := mkMockDb(t, ms)
+	conv := internal.MakeConv()
+	err := ProcessInfoSchema(conv, db)
+	assert.Nil(t, err)
+	expectedSchema := map[string]ddl.CreateTable{
+		"cart": ddl.CreateTable{
+			Name:     "cart",
+			ColNames: []string{"productid", "userid", "quantity"},
+			ColDefs: map[string]ddl.ColumnDef{
+				"productid": ddl.ColumnDef{Name: "productid", T: ddl.String{Len: ddl.MaxLength{}}, NotNull: true},
+				"userid":    ddl.ColumnDef{Name: "userid", T: ddl.String{Len: ddl.MaxLength{}}, NotNull: true},
+				"quantity":  ddl.ColumnDef{Name: "quantity", T: ddl.Int64{}},
+			},
+			Pks: []ddl.IndexKey{ddl.IndexKey{Col: "productid"}, ddl.IndexKey{Col: "userid"}}},
+		"test": ddl.CreateTable{
+			Name:     "test",
+			ColNames: []string{"id", "s", "txt", "b", "bs", "bl", "c", "c8", "d", "f8", "f4", "i8", "i4", "i2", "si", "ts", "tz", "vc", "vc6"},
+			ColDefs: map[string]ddl.ColumnDef{
+				"id":  ddl.ColumnDef{Name: "id", T: ddl.Int64{}, NotNull: true},
+				"s":   ddl.ColumnDef{Name: "s", T: ddl.String{Len: ddl.MaxLength{}}, IsArray: true},
+				"txt": ddl.ColumnDef{Name: "txt", T: ddl.String{Len: ddl.MaxLength{}}, NotNull: true},
+				"b":   ddl.ColumnDef{Name: "b", T: ddl.Bool{}},
+				"bs":  ddl.ColumnDef{Name: "bs", T: ddl.Int64{}, NotNull: true},
+				"bl":  ddl.ColumnDef{Name: "bl", T: ddl.Bytes{Len: ddl.MaxLength{}}},
+				"c":   ddl.ColumnDef{Name: "c", T: ddl.String{Len: ddl.Int64Length{Value: 1}}},
+				"c8":  ddl.ColumnDef{Name: "c8", T: ddl.String{Len: ddl.Int64Length{Value: 8}}},
+				"d":   ddl.ColumnDef{Name: "d", T: ddl.Date{}},
+				"f8":  ddl.ColumnDef{Name: "f8", T: ddl.Float64{}},
+				"f4":  ddl.ColumnDef{Name: "f4", T: ddl.Float64{}},
+				"i8":  ddl.ColumnDef{Name: "i8", T: ddl.Int64{}},
+				"i4":  ddl.ColumnDef{Name: "i4", T: ddl.Int64{}},
+				"i2":  ddl.ColumnDef{Name: "i2", T: ddl.Int64{}},
+				"si":  ddl.ColumnDef{Name: "si", T: ddl.Int64{}, NotNull: true},
+				"ts":  ddl.ColumnDef{Name: "ts", T: ddl.Timestamp{}},
+				"tz":  ddl.ColumnDef{Name: "tz", T: ddl.Timestamp{}},
+				"vc":  ddl.ColumnDef{Name: "vc", T: ddl.String{Len: ddl.MaxLength{}}},
+				"vc6": ddl.ColumnDef{Name: "vc6", T: ddl.String{Len: ddl.Int64Length{Value: 6}}},
+			},
+			Pks: []ddl.IndexKey{ddl.IndexKey{Col: "id"}}},
+	}
+	assert.Equal(t, expectedSchema, stripSchemaComments(conv.SpSchema))
+	assert.Equal(t, len(conv.Issues["cart"]), 0)
+	expectedIssues := map[string][]internal.SchemaIssue{
+		"bs": []internal.SchemaIssue{internal.DefaultValue},
+		"f4": []internal.SchemaIssue{internal.Widened},
+		"i4": []internal.SchemaIssue{internal.Widened},
+		"i2": []internal.SchemaIssue{internal.Widened},
+		"si": []internal.SchemaIssue{internal.Widened, internal.DefaultValue},
+		"ts": []internal.SchemaIssue{internal.Datetime},
+	}
+	assert.Equal(t, expectedIssues, conv.Issues["test"])
+	assert.Equal(t, int64(0), conv.Unexpecteds())
+}
+
+// TODO : Remove this test
```

Comment:
if this is not needed, remove this test

Reply:
oh I had added that comment for Nevin/Hengfeng. Because we have changed the way data conversion is handled and clubbed it for mysqldump and mysql_infoschema. Let it be there, they can take a call if they want to keep it or not because this test is there in postgres_infoschema.